### PR TITLE
interface\patient_file\report library refactor

### PR DESCRIPTION
**Pull request by MigDinny**
#915 

Refactor on library -> now using call_required_libraries

jquery was upgraded to the latest version

### CODE REVIEW

File: library/headers.inc.php [hunk 1]
```diff
@@ -102,6 +129,4 @@ function resolveFancyboxCompatibility(){ ?>
 <?php 
 } 
 //The following seem to always get called.  I don't believe this is proper.
-?>
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/textformat.js"></script>
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/dialog.js"></script>
+?>
```
Comment (aethelwulffe) [inline library/headers.inc.php]: Newline please.
Personally, I like if(){  vs if() {
-Not really clean to me, but it is fine.

**Comment by aethelwulffe** [comment]
@MigDinny   Miguel, add this new line real quick so I can approve it.  We are trying to get some things merged up here.

**Comment by MigDinny** [comment]
@aethelwulffe Sorry I was working on other stuff. 1 sec

File: library/headers.inc.php (hunk 1 above)
Comment (MigDinny) [reply library/headers.inc.php]: I prefer if() {

I think this is the kind of stuff we get used to that our mind will never change

**Comment by MigDinny** [comment]
@aethelwulffe is it good now?

File: library/headers.inc.php (hunk 1 above)
Comment (teryhill) [reply library/headers.inc.php]: I prefer if ()
             {

             } else {

             }

Comment (MigDinny) [reply library/headers.inc.php]: :(

Comment (aethelwulffe) [reply library/headers.inc.php]: :)

**Comment by aethelwulffe** [comment]
@tmccormi @teryhill @Trodrige @pri2si17-1997 

Someone else please test and review.  We have a pile of JS version refactor PR's here we should be proceeding on.  This is the first one to be validated.

File: interface/patient_file/summary/demographics.php [hunk 2]
```diff
@@ -1,1687 +1,1638 @@
...
+  }
+  
+  // Display image in 'widget style'
+  function image_widget($doc_id,$doc_catg)
+  {
+          global $pid, $web_root;
+          $docobj = new Document($doc_id);
+          $image_file = $docobj->get_url_file();
+          $extension = substr($image_file, strrpos($image_file,"."));
+          $viewable_types = array('.png','.jpg','.jpeg','.png','.bmp','.PNG','.JPG','.JPEG','.PNG','.BMP'); // image ext supported by fancybox viewer
```
Comment (aethelwulffe) [inline interface/patient_file/summary/demographics.php]: $finfo = new finfo;
 
$fileinfo = $finfo->file($file, FILEINFO_MIME);
Should use fileinfo extension

**Comment by pri2si17-1997** [comment]
Hello @aethelwulffe @teryhill I tested it and it is working fine. I was not able to find any error, but I doubt merging this pr and removal of `html_header_show()` .

**Comment by aethelwulffe** [comment]
@pri2si17-1997 What are your specific doubts on this particular PR?

**Comment by pri2si17-1997** [comment]
Hello @aethelwulffe I have no specific doubt with this PR. I was thinking merging this and `html_header_show()` PR, may not leave some file with functions contained in it.

**Comment by aethelwulffe** [comment]
You think that there might be conflicts or reconstitution of files or something?  Well, nothing to do other than simulate the merges and find out.
@pri2si17-1997 Do you, or do you not have code review privileges?

**Comment by MigDinny** [comment]
Conflicts solved. I think there is no need to test files again because I've touched only headers.inc.php

But take a look just to check if this commit is ok to go to the codebase

**Comment by aethelwulffe** [comment]
This looks ready to go.

**Comment by aethelwulffe** [comment]
This PR is ready for second review.  ATTENTION!

**Pull request merged**